--- a/net/netfilter/nf_conntrack_ecache.c
+++ b/net/netfilter/nf_conntrack_ecache.c
@@ -123,8 +123,9 @@
 	if (delay >= 0)
 		schedule_delayed_work(&cnet->ecache.dwork, delay);
 }
 
++#ifndef CONFIG_NF_CONNTRACK_CHAIN_EVENTS
 static int __nf_conntrack_eventmask_report(struct nf_conntrack_ecache *e,
 					   const u32 events,
 					   const u32 missed,
 					   const struct nf_ct_event *item)
@@ -160,8 +161,9 @@
 	} while (cmpxchg(&e->missed, old, want) != old);
 
 	return ret;
 }
++#endif
 
 int nf_conntrack_eventmask_report(unsigned int events, struct nf_conn *ct,
 				  u32 portid, int report)
 {
